src/api/commands/currencyprice/CurrencyPriceRootCommand.ts

```diff
@@ -4,30 +4,50 @@ import { RpcCommandInterface } from '../RpcCommandInterface';
 import { validate, request } from '../../../core/api/Validate';
 import { Logger as LoggerType } from '../../../core/Logger';
 import { Types, Core, Targets } from '../../../constants';
-import { CurrencyPriceAddCommand } from './CurrencyPriceAddCommand';
 import { BaseCommand } from '../BaseCommand';
 import { RpcCommandFactory } from '../../factories/RpcCommandFactory';
 import { Commands } from '../CommandEnumType';
+import { CurrencyPriceService } from '../../services/CurrencyPriceService';
+import { MessageException } from '../../exceptions/MessageException';
 
-export class CurrencyPriceRootCommand extends BaseCommand implements RpcCommandInterface<void> {
+export class CurrencyPriceRootCommand extends BaseCommand implements RpcCommandInterface<any> {
 
     public log: LoggerType;
 
     constructor(
-        @inject(Types.Command) @named(Targets.Command.currencyprice.CurrencyPriceAddCommand) private currencyPriceAddCommand: CurrencyPriceAddCommand,
+        @inject(Types.Service) @named(Targets.Service.CurrencyPriceService) private currencyPriceService: CurrencyPriceService,
         @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
     ) {
         super(Commands.CURRENCYPRICE_ROOT);
         this.log = new Logger(__filename);
     }
 
+    /**
+     *
+     * data.params[]:
+     * [0]: fromCurrency
+     * [1]: toCurrency
+     * [...]: toCurrency
+     *
+     * description: fromCurrency must be PART for now and toCurrency may be multiple currencies like INR, USD etc..
+     * example: [PART, INR, USD, EUR, GBP, ....]
+     */
     @validate()
     public async execute( @request(RpcRequest) data: RpcRequest, rpcCommandFactory: RpcCommandFactory): Promise<any> {
-        return await this.executeNext(data, rpcCommandFactory);
+        const fromCurrency = data.params.shift();
+        // throw exception if fromCurrency is not a PART or toCurrencies has length 0
+        if (fromCurrency !== 'PART' || data.params.length < 1 ) {
+           throw new MessageException('Invalid params');
+        } else {
+           return await this.currencyPriceService.getCurrencyPrices(fromCurrency, data.params);
+        }
     }
 
     public help(): string {
-        return this.getName() + ' (add)';
+        return this.getName() + '<from>, ,[to...])\n'
+            + '    <from>                     - String - The currency name from we want to convert the currency\n'
+
+            + '    <to>                       - Sting[] - Array of currencies name to we want to convert the currency\n';
     }
 
     public description(): string {
```

Comment:
we're not managing any currency converter here, it's just a Command to convert currencies